Mathlib/AlgebraicGeometry/Morphisms/Basic.lean

```diff
@@ -156,347 +233,248 @@ instance targetAffineLocally_respectsIso (P : AffineTargetMorphismProperty)
-        ∃ 𝒰 : Scheme.OpenCover.{u} Y,
-          ∀ i : 𝒰.J, P (pullback.snd f (𝒰.map i)),
-        ∀ (𝒰 : Scheme.OpenCover.{u} Y) (i : 𝒰.J),
-          P (pullback.snd f (𝒰.map i)),
-        ∀ U : Opens Y.carrier, P (f ∣_ U),
-        ∀ {U : Scheme} (g : U ⟶ Y) [IsOpenImmersion g], P (pullback.snd f g),
-        ∃ (ι : Type u) (U : ι → Opens Y.carrier) (_ : iSup U = ⊤), ∀ i, P (f ∣_ U i)] := by
-  have := hP.1
-  tfae_have 2 → 1
-  · rintro ⟨𝒰, H⟩; exact hP.3 f 𝒰 H
-  tfae_have 1 → 4
-  · intro H U; exact hP.2 f U H
-  tfae_have 4 → 3
-  · intro H 𝒰 i
-    rw [← P.arrow_mk_iso_iff (morphismRestrictOpensRange f _)]
-    exact H <| Scheme.Hom.opensRange (𝒰.map i)
-  tfae_have 3 → 2
-  · exact fun H => ⟨Y.affineCover, H Y.affineCover⟩
-  tfae_have 4 → 5
-  · intro H U g hg
-    rw [← P.arrow_mk_iso_iff (morphismRestrictOpensRange f _)]
-    apply H
-  tfae_have 5 → 4
-  · intro H U
-    erw [P.cancel_left_of_respectsIso]
-    apply H
-  tfae_have 4 → 6
-  · intro H; exact ⟨PUnit, fun _ => ⊤, ciSup_const, fun _ => H _⟩
-  tfae_have 6 → 2
-  · rintro ⟨ι, U, hU, H⟩
-    refine ⟨Y.openCoverOfSuprEqTop U hU, ?_⟩
-    intro i
-    rw [← P.arrow_mk_iso_iff (morphismRestrictOpensRange f _)]
-    convert H i
-    all_goals ext1; exact Subtype.range_coe
-  tfae_finish
-#align algebraic_geometry.property_is_local_at_target.open_cover_tfae AlgebraicGeometry.PropertyIsLocalAtTarget.openCover_TFAE
-
-theorem PropertyIsLocalAtTarget.openCover_iff {P : MorphismProperty Scheme}
-    (hP : PropertyIsLocalAtTarget P) {X Y : Scheme.{u}} (f : X ⟶ Y) (𝒰 : Scheme.OpenCover.{u} Y) :
-    P f ↔ ∀ i, P (pullback.snd f (𝒰.map i)) := by
-  -- Porting note: couldn't get the term mode proof work
-  refine ⟨fun H => let h := ((hP.openCover_TFAE f).out 0 2).mp H; fun i => ?_,
-    fun H => let h := ((hP.openCover_TFAE f).out 1 0).mp; ?_⟩
-  · exact h 𝒰 i
-  · exact h ⟨𝒰, H⟩
-#align algebraic_geometry.property_is_local_at_target.open_cover_iff AlgebraicGeometry.PropertyIsLocalAtTarget.openCover_iff
-
-namespace AffineTargetMorphismProperty
+open AffineTargetMorphismProperty in
+instance (P : MorphismProperty Scheme) [IsLocalAtTarget P] : (of P).IsLocal where
+  respectsIso := inferInstance
+  to_basicOpen _ _ H := IsLocalAtTarget.restrict H _
+  of_basicOpenCover {_ Y} _ _ _ hs := IsLocalAtTarget.of_iSup_eq_top _
+    (((isAffineOpen_top Y).basicOpen_union_eq_self_iff _).mpr hs)
 
 /-- A `P : AffineTargetMorphismProperty` is stable under base change if `P` holds for `Y ⟶ S`
 implies that `P` holds for `X ×ₛ Y ⟶ X` with `X` and `S` affine schemes. -/
-def StableUnderBaseChange (P : AffineTargetMorphismProperty) : Prop :=
+def AffineTargetMorphismProperty.StableUnderBaseChange (P : AffineTargetMorphismProperty) : Prop :=
   ∀ ⦃X Y S : Scheme⦄ [IsAffine S] [IsAffine X] (f : X ⟶ S) (g : Y ⟶ S),
     P g → P (pullback.fst f g)
 #align algebraic_geometry.affine_target_morphism_property.stable_under_base_change AlgebraicGeometry.AffineTargetMorphismProperty.StableUnderBaseChange
```

Comment:
Side note: Should this be defined in terms of `IsPullback` mimicking `MorphismProperty.StableUnderBaseChange`?